referenceframe/frame.go

```diff
@@ -630,3 +665,83 @@ func PoseToInputs(p spatial.Pose) []Input {
 		p.Orientation().OrientationVectorRadians().Theta,
 	})
 }
+
+// Determine whether two frames are (nearly) identical. For now, we only support implementers of
+// the frame interface that are registered (see register.go). However, this is not automatic, so
+// if a new implementer is registered manually in register.go, its case should be added here.
+//
+// NOTE: for ease, this function only takes one epsilon parameter because we have yet
+// to see a case of quantity where we want accept different levels of floating point error.
+// If the time comes where we want a different allowance for limits, vectors, and geometries,
+// this function should be changed accordingly.
+func framesAlmostEqual(frame1, frame2 Frame, epsilon float64) (bool, error) {
+	switch {
+	case reflect.TypeOf(frame1) != reflect.TypeOf(frame2):
+		return false, nil
+	case frame1.Name() != frame2.Name():
+		return false, nil
+	case !limitsAlmostEqual(frame1.DoF(), frame2.DoF(), epsilon):
+		return false, nil
+	default:
+	}
+
+	switch f1 := frame1.(type) {
+	case *staticFrame:
+		f2 := frame2.(*staticFrame)
+		switch {
+		case !spatial.PoseAlmostEqual(f1.transform, f2.transform):
+			return false, nil
+		case f1.geometry == nil && f2.geometry == nil:
+			return true, nil
+		case !spatial.GeometriesAlmostEqual(f1.geometry, f2.geometry):
```

Comment:
doesn't this panic if only one of the geometries are nil but the other isn't nil?